simSPI/linear_simulator/modules.py

```diff
@@ -0,0 +1,397 @@
+"""contains simulator and its components classes."""
+import os
+
+import mrcfile
+import numpy as np
+import torch
+import torch.fft
+from pytorch3d.transforms import Rotate
+
+from .utils import fourier_to_primal_2D, init_cube, primal_to_fourier_2D
+
+"""Module to generate data using using liner forward model."""
+
+
+class simulator(torch.nn.Module):
+    """Class to generate data using liner forward model.
+
+    Parameters
+    ----------
+    config: class
+        Class containing parameters of the simulator
+
+    """
+
+    def __init__(self, config, initial_volume=None):
+        super(simulator, self).__init__()
+
+        self.config = config
+        self.Projector = Projector(config)  # class for tomographic projector
+        self.init_volume()  # changes the volume inside the projector
+        self.CTF = CTF(config)  # class for ctf
```

Comment:
I'm not exactly sure what naming case this falls into https://www.python.org/dev/peps/pep-0008/#naming-conventions

Are you sure it should be 
```
self.Projector = Projector(config)
...
self.CTF = CTF(config)
```
and not 
```
self.projector = Projector(config)
self.ctf = CTF(config)
```

Reply:
Nevermind, I see how these are still Classes, with there own methods that need to be called.

Reply:
@geoffwoollard I think you were correct earlier. it should be `self.projector` as `projector` is a class instance variable not a class name.

Reply:
You are right. Thanks! These names have been/will be changed according to the python convention in the new PRs.